core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -824,6 +827,15 @@ public NativeImageInvokerInfo build() {
                 if (nativeConfig.enableVmInspection) {
                     nativeImageArgs.add("-H:+AllowVMInspection");
                 }
+                if (nativeConfig.monitoring.isPresent()) {
+                    nativeImageArgs
+                            .add("--enable-monitoring=" + nativeConfig.monitoring.get().stream().map(o -> o.name().toLowerCase(
+                                    Locale.ROOT)).collect(Collectors.joining(",")));
+                } else if (ConfigProvider.getConfig().getConfigValue("quarkus.native.monitoring").getValue() != null) {
+                    // this only happens when a user has configured 'quarkus.native.monitoring='
```

Comment:
This holds for `-Dquarkus.native.monitoring` as well, right?

Reply:
You need `-Dquarkus.native.monitoring=` because `-Dquarkus.native.monitoring` is considered equivalent to `-Dquarkus.native.monitoring=true`

Reply:
I think handling `-Dquarkus.native.monitoring` is more important than handling `-Dquarkus.native.monitoring=`.
Especially for people expecting it to work because of the way that `--enable-monitoring` is supported in GraalVM.
Do you think it's worth adding a check for it (i.e., if `true` pass `--enable-monitoring` or `--enable-monitoring=all`)?

Reply:
I don't think it's terribly important but nonetheless I oblige.

Reply:
GraalVM is deprecating `--enable-monitoring` without arguments... https://github.com/oracle/graal/pull/5792 :facepalm: 

I will prepare a PR to drop the support for it...

Reply:
😂